lib/src/system.dart

```diff
@@ -0,0 +1,45 @@
+import 'dart:io';
+
+import 'package:glob/glob.dart';
+import 'package:glob/list_local_fs.dart';
+import 'package:path/path.dart' as p;
+
+import 'collections.dart';
+import 'objects.dart';
+import 'strings.dart';
+
+// ignore: public_member_api_docs
+typedef FileSystemEntityPeekFunction = void Function(FileSystemEntity element);
+
+/// Deletes all [FileSystemEntity]s matching the provided globs.
+///
+/// The globs will be combined to `Match one of several possibilities {...,...}`
+///
+/// See https://pub.dev/packages/glob for more information about globs
+///
+/// [workingDirectory] will be the root path for the glob pattern to be
+/// evaluated in. Defaults to the current working directory provided by
+/// [p.context]
+///
+/// `since 0.0.1`
+Future<void> deleteFilesMatchingGlobs(
+  Iterable<String> globs, {
+  FileSystemEntityPeekFunction? peek,
+  String? workingDirectory,
+}) async {
+  if (isEmpty(globs)) {
+    return;
+  }
+  var globContext = isNotBlank(workingDirectory)
+      ? p.Context(current: workingDirectory)
+      : p.context;
+  await Glob(
+    "{${globs.join(',')}}",
+    context: globContext,
+  ).list(root: globContext.current).forEach((fse) async {
+    if (isNotNull(peek)) {
+      peek!(fse);
+    }
```

Comment:
I think we should remove `peek` and instead have two functions: one that does the globbing (returning the `Stream<FileSystemEntity>`), and one that does the delete. Makes the function better since then each function is then only doing one thing.

Reply:
Sure.